meson.build

```
@@ -380,30 +380,52 @@ luaold_interpreters = [
 ]
 
 lua_pc_name = ''
+have_elua = get_option('elua')
 
-if sys_windows or get_option('lua-interpreter') == 'lua'
-  if sys_windows and get_option('lua-interpreter') != 'lua'
-    warning('There is no "' + get_option('lua-interpreter')
-            + '" binding option for Windows, backing up to "lua"')
-  endif
+if get_option('lua-interpreter') == 'lua'
   config_h.set('ENABLE_LUA_OLD', '1')
   foreach l : luaold_interpreters
     lua = dependency(l[0], version: l[1], required:false)
     lua_pc_name = l[0]
-    if lua.found() == true
+    if lua.found()
       break
     endif
   endforeach
+  if not lua.found()
+    lua = dependency('lua', fallback : ['lua', 'lua_dep'])
+    if not lua.found()
+      error('Lua not found')
+    endif
+  endif
+  if have_elua
+    luaver_min = cc.compute_int('LUA_VERSION_NUM - 500',
+      prefix: '#include <lua.h>', dependencies: lua
+    )
+    lua_ffi = dependency('cffi-lua-5.@0@'.format(luaver_min), required: false)
+    if not lua_ffi.found()
+      error('Elua with interpreter is experimental, disable it or install cffi-lua...')
+    else
+      message('Using experimental Elua with interpreter support...')
+    endif
+  endif
 else
   lua = dependency(get_option('lua-interpreter'))
   lua_pc_name = 'luajit'
+  if not lua.found()
+    lua = dependency('luajit')
+    if not lua.found()
+      error('Lua not found')
+    endif
+  endif
```

As for now we don't have a `luajit` fallback, so I think we don't need to double check it:
```suggestion
else
  lua = dependency(get_option('lua-interpreter'))
  lua_pc_name = 'luajit'
  if not lua.found()
    error('Lua not found')
  endif
```